problem_builder/public/js/mentoring_with_steps.js

```diff
@@ -7,30 +7,118 @@ function MentoringWithStepsBlock(runtime, element) {
     }
 
     var children = runtime.children(element);
+
     var steps = [];
 
     for (var i = 0; i < children.length; i++) {
         var child = children[i];
         var blockType = $(child.element).data('block-type');
         if (blockType === 'sb-step') {
-            steps.push(child);
+            registerStep(child);
         }
     }
 
-    var activeStep = $('.mentoring', element).data('active-step');
+    var activeStepIndex = $('.mentoring', element).data('active-step');
     var attemptsTemplate = _.template($('#xblock-attempts-template').html());
     var message = $('.sb-step-message', element);
     var checkmark, submitDOM, nextDOM, reviewButtonDOM, tryAgainDOM,
         gradeDOM, attemptsDOM, reviewLinkDOM, submitXHR;
     var reviewStepDOM = $("div.xblock[data-block-type=sb-review-step], div.xblock-v1[data-block-type=sb-review-step]", element);
     var hasAReviewStep = reviewStepDOM.length == 1;
+    
+    /**
+     * Registers step to in this builder, it stores the xblock element, and some additional metadata used to safely
+     * remove and re-attach xblock to the DOM. See showStep and hideStep to see why we need to do it.
```

Comment:
@jbzdak "Registers step with this StepBuilder instance: Creates and stores a wrapper that contains the XBlock instance and associated DOM element, as well as some additional metadata ..."

Very happy to see documentation for all of the new functions! :)